File: 5_percent/07_10001st_prime.py
import _common_functions as cf

def numberth_prime(number):
  """
  number: index of all the prime numbers
  return: the numberth prime number and its index, in this case number
  """
  i = 0
  n = 0
  while i < number:
      n += 1
      if cf.is_prime(n):
          i += 1
  return n

assert numberth_prime(10001) == 104743
print(cf.average_running_time(10, numberth_prime, 10001))

File: 5_percent_c/07_10001st_prime.c
#include <stdio.h>

int isPrime(int number)
{
	for(int i = 2; i<(number/2)+1; i++)
	{
		if (number % i == 0)
			return 0;
	}
	return 1;
}

int main()
{
	int number = 1;
	int prime_count = 0;
	while(prime_count < 10001)
	{
		number++;
		prime_count += isPrime(number);
	}
	printf("the 10001st prime is %d\n", number);

	return 0;
}
